pytraj/all_actions.py

```diff
@@ -157,6 +158,55 @@ def _assert_mutable(trajiter):
             "This analysis does not support immutable object. Use `pytraj.Trajectory`"
         )
 
+def pair_distance(p1, p2):
+    x1, y1, z1 = p1
+    x2, y2, z2 = p2
+    return np.sqrt((x1 - x2)**2 + (y1 - y2)**2 + (z1 - z2)**2)
+
+@register_pmap
+def closest_atom(top=None, frame=None, point=(0, 0, 0), mask=""):
```

Comment:
- it seems `top` and `frame` are required. So I think they should be positional argument, not key word argument

`def closest_atom(top, frame, point=(0, 0, 0), mask="")`

- is `mask` optional?

- A design question, depending on your need.

Which one is good for you?
```
closest_atom(top, frame, ...)
closest_atom(traj, frame_index: int, ...)
```

Reply:
Changed the top/frame to positional. Mask is optional, I want it to default to all atoms but I have mostly been passing water atoms into the mask for aiding GIST analysis.

I prefer the former, for some reason this seems like a frame operation to me and not a trajectory operation. I can change it if you disagree though.

Reply:
>  I can change it if you disagree though.

I don't have a use case for the function but you do have one. So let's keep your original approach. Cheers.